### PR TITLE
build: manual usage of semgrep instead of github-actions

### PR DESCRIPTION
**Pull request by DawoudSheraz**
### Description
- Use semgrep manually instead of GitHub action. This allows using the latest version of semgrep that will potentially fix semgrep issues on various PRs. Base PR: https://github.com/openedx/course-discovery/pull/3505

- With selenium 4.3.0, find_elements_by_* methods have been removed. https://github.com/SeleniumHQ/selenium/pull/10712.  The correct method to use is find_element/find_elements where the first argument is search criteria (id, xpath, css selector, etc.). https://selenium-python.readthedocs.io/locating-elements.html

### CODE REVIEW

File: requirements/test.in [hunk 1]
```diff
@@ -16,6 +16,7 @@ pytest-django
 pytest-responses
 pytest-xdist
 responses
+semgrep
```
Comment (Ali-D-Akbar) [inline requirements/test.in:19]: It doesn't seem like this package added anything while running the upgrade, did it?

Comment (DawoudSheraz) [reply requirements/test.in:19]: It did, in local.txt. The requirement files' structure for discovery is a weird one. There is no test.txt. Instead, test.in is added in local.in.

**Comment by nedbat** [comment]
I'm not sure why you didn't need to make the pylint fixes that I made in #3505, but these tests are passing, so :+1:

**Review by nedbat — APPROVED** [review]

**Comment by DawoudSheraz** [comment]
> I'm not sure why you didn't need to make the pylint fixes that I made in #3505, but these tests are passing, so 👍

Those fixes were done in https://github.com/openedx/course-discovery/pull/3498 which was just merged today.

**Review by AliAdnanSohail — APPROVED** [review]

**Pull request merged**